### ISSUE TITLE
Allow multiple activationEvents to be met before activating an extension

### ISSUE THREAD

**Issue by Septh** [Microsoft/vscode#62367]
Issue Type: <b>Feature Request</b>

**> Context**
I am thinking about refactoring my [wow-bundle](https://github.com/Septh/vscode-wow-bundle) extension.

In its current form, wow-bundle overwrites vscode's builtin Lua grammar with a WoW-specific one, which is far from ideal for people who need to edit Lua files that are not part of a World of Warcraft addon. 

So one of my main concerns is activating the extension only when needed:

	"activationEvents": [
		"onLanguage:toc",
		"onLanguage:lua",
		"workspaceContains:/*.toc"
	]

(the .toc file being to a WoW addon what package.json is to a vscode extension)

**> Proposal**
This works fine but with this scheme, the extension will activate when *any one* condition is met. What I'd like is a way to activate the extension when *multiple* conditions are met, ie., something like this:

	"activationEvents": [
		"onLanguage:toc",
		[ 
                      "onLanguage:lua",
                      "workspaceContains:/*.toc" 
		]
	]

which would mean: activate the extension when either (a .toc file is edited) **OR** (a .lua file is edited **AND** the containing folder has a .toc file at its root). 

In short: grouping activation events in an array would make them mandatory. And this syntax would maintain compatibility with current extensions.

**> Alternate proposal**
An other way would be to allow an extension *not* to load, eg. by having its activate() function return some special value, presumably `null`:

	export function activate(context: vscode.ExtensionContext) {
		if ( conditions_are_not_met(context) ) {
			return null
		}
	}

For this to be really useful, the `ExtensionContext` should be extended with an `activationEvent` field which would allow the extension to know what event activated it.

What do you think?

VS Code version: Code 1.28.2 (7f3ce96ff4729c91352ae6def877e59c561f4850, 2018-10-17T00:23:51.859Z)
OS version: Windows_NT x64 10.0.17134

**Comment by Septh** [comment]
Well, after some digging in the docs, it seems that I misunderstood the point of the activate() method... My bad. ;)

**Comment by alexandrudima** [comment]
@Septh Yes, contributions are always applied, irrespective of `activationEvents`. That one is only for the case where your extension writes code.
